### ISSUE TITLE
Example not working on minikube 1.14

### ISSUE THREAD

**Issue by ervikrant06** [kubernetes-csi/csi-driver-host-path#53]
I came to this repo while reading https://kubernetes.io/blog/2019/03/07/raw-block-volume-support-to-beta/ I can't understand the purpose of mixing hostpath and raw block volumes. can you please help me to understand it?

- Ran the script for 1.14
~~~
$ kubectl get pod
NAME                         READY   STATUS              RESTARTS   AGE
csi-hostpath-attacher-0      1/1     Running             0          6m2s
csi-hostpath-provisioner-0   1/1     Running             0          6m1s
csi-hostpath-snapshotter-0   1/1     Running             0          6m
csi-hostpath-socat-0         1/1     Running             0          5m59s
csi-hostpathplugin-0         1/3     RunContainerError   0          6m2s
~~~

- Following error reported in the POD Logs. 

~~~
Events:
  Type     Reason     Age    From               Message
  ----     ------     ----   ----               -------
  Normal   Scheduled  5m7s   default-scheduler  Successfully assigned default/csi-hostpathplugin-0 to minikube
  Normal   Pulling    5m4s   kubelet, minikube  Pulling image "quay.io/k8scsi/csi-node-driver-registrar:v1.1.0"
  Normal   Pulled     4m54s  kubelet, minikube  Successfully pulled image "quay.io/k8scsi/csi-node-driver-registrar:v1.1.0"
  Normal   Created    4m53s  kubelet, minikube  Created container node-driver-registrar
  Normal   Started    4m53s  kubelet, minikube  Started container node-driver-registrar
  Normal   Pulling    4m53s  kubelet, minikube  Pulling image "quay.io/k8scsi/hostpathplugin:v1.1.0"
  Normal   Pulled     3m47s  kubelet, minikube  Successfully pulled image "quay.io/k8scsi/hostpathplugin:v1.1.0"
  Normal   Created    3m46s  kubelet, minikube  Created container hostpath
  Warning  Failed     3m46s  kubelet, minikube  Error: failed to start container "hostpath": Error response from daemon: OCI runtime create failed: open /var/run/docker/runtime-runc/moby/hostpath/state.json: no such file or directory: unknown
  Normal   Pulling    3m46s  kubelet, minikube  Pulling image "quay.io/k8scsi/livenessprobe:v1.1.0"
  Warning  Failed     3m45s  kubelet, minikube  Failed to pull image "quay.io/k8scsi/livenessprobe:v1.1.0": rpc error: code = Unknown desc = Error response from daemon: Get https://quay.io/v2/: dial tcp: lookup quay.io on [::1]:53: read udp [::1]:42931->[::1]:53: read: connection refused
  Warning  Failed     3m45s  kubelet, minikube  Error: ErrImagePull
~~~

**Comment by pohly** [comment]
ervikrant06 <notifications@github.com> writes:

> I came to this repo while reading
> https://kubernetes.io/blog/2019/03/07/raw-block-volume-support-to-beta/
> I can't understand the purpose of mixing hostpath and raw block
> volumes.

You don't need the hostpath driver for raw block volumes. It's just an
example driver which supports them.

>   Warning  Failed     3m46s  kubelet, minikube  Error: failed to start container "hostpath": Error response from daemon: OCI runtime create failed: open /var/run/docker/runtime-runc/moby/hostpath/state.json: no such file or directory: unknown
>   Normal   Pulling    3m46s  kubelet, minikube  Pulling image "quay.io/k8scsi/livenessprobe:v1.1.0"
>   Warning  Failed     3m45s  kubelet, minikube  Failed to pull image "quay.io/k8scsi/livenessprobe:v1.1.0": rpc error: code = Unknown desc = Error response from daemon: Get https://quay.io/v2/: dial tcp: lookup quay.io on [::1]:53: read udp [::1]:42931->[::1]:53: read: connection refused
>   Warning  Failed     3m45s  kubelet, minikube  Error: ErrImagePull

This looks like a temporary network error. The other images are also on
quay.io and were pulled just fine.

**Comment by moonlight16** [comment]
Is this the same issue?

```
$ kubectl get -w pods
NAME                         READY   STATUS              RESTARTS   AGE
csi-hostpath-attacher-0      1/1     Running             0          87s
csi-hostpath-provisioner-0   1/1     Running             0          86s
csi-hostpath-snapshotter-0   1/1     Running             0          85s
csi-hostpath-socat-0         1/1     Running             0          84s
csi-hostpathplugin-0         1/3     RunContainerError   0          86s
web-0                        1/1     Running             0          6d5h
web-1                        1/1     Running             0          6d5h

```

```
Events:
  Type     Reason     Age    From               Message
  ----     ------     ----   ----               -------
  Normal   Scheduled  5m14s  default-scheduler  Successfully assigned default/csi-hostpathplugin-0 to minikube
  Normal   Pulling    5m13s  kubelet, minikube  Pulling image "quay.io/k8scsi/csi-node-driver-registrar:v1.1.0"
  Normal   Pulled     5m10s  kubelet, minikube  Successfully pulled image "quay.io/k8scsi/csi-node-driver-registrar:v1.1.0"
  Normal   Created    5m10s  kubelet, minikube  Created container node-driver-registrar
  Normal   Started    5m10s  kubelet, minikube  Started container node-driver-registrar
  Normal   Pulling    5m10s  kubelet, minikube  Pulling image "quay.io/k8scsi/hostpathplugin:v1.2.0-rc6"
  Normal   Pulled     4m52s  kubelet, minikube  Successfully pulled image "quay.io/k8scsi/hostpathplugin:v1.2.0-rc6"
  Normal   Created    4m52s  kubelet, minikube  Created container hostpath
  Warning  Failed     4m52s  kubelet, minikube  Error: failed to start container "hostpath": Error response from daemon: OCI runtime create failed: open /var/run/docker/runtime-runc/moby/hostpath/state.json: no such file or directory: unknown
  Normal   Pulling    4m52s  kubelet, minikube  Pulling image "quay.io/k8scsi/livenessprobe:v1.1.0"
  Warning  Failed     4m     kubelet, minikube  Failed to pull image "quay.io/k8scsi/livenessprobe:v1.1.0": rpc error: code = Unknown desc = dial tcp: lookup quay.io on [::1]:53: read udp [::1]:41202->[::1]:53: read: connection refused
  Warning  Failed     4m     kubelet, minikube  Error: ErrImagePull
```

**Comment by msau42** [comment]
Error message looks similar to me.  @kfox1111 were you able to make any headway on supporting CSI in minikube?

**Comment by kfox1111** [comment]
This is fixed in minikube v1.4.0.
